File: src/main/java/extracells/gui/GuiHardMEDrive.java
package extracells.gui;

import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.inventory.GuiContainer;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.util.ResourceLocation;
import net.minecraft.world.World;

import org.lwjgl.opengl.GL11;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import extracells.BlockEnum;
import extracells.container.ContainerHardMEDrive;
import extracells.tile.TileEntityHardMEDrive;

@SideOnly(Side.CLIENT)
public class GuiHardMEDrive extends GuiContainer
{

	public static final int xSize = 176;
	public static final int ySize = 166;

	public GuiHardMEDrive(InventoryPlayer inventory, TileEntityHardMEDrive tileentity)
	{
		super(new ContainerHardMEDrive(inventory, tileentity));
	}

	@Override
	protected void drawGuiContainerBackgroundLayer(float f, int i, int j)
	{
		drawDefaultBackground();
		GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
		Minecraft.getMinecraft().renderEngine.bindTexture(new ResourceLocation("extracells", "textures/gui/hardmedrive.png"));
		int posX = (width - xSize) / 2;
		int posY = (height - ySize) / 2;
		drawTexturedModalRect(posX, posY, 0, 0, xSize, ySize);
	}

	@Override
	protected void drawGuiContainerForegroundLayer(int i, int j)
	{
		this.fontRenderer.drawString(BlockEnum.BLASTRESISTANTMEDRIVE.getStatName(), 5, 5, 0x000000);
	}

	public int getRowLength()
	{
		return 1;
	}
}

File: src/main/scala/extracells/gui/GuiHardMEDrive.scala
package extracells.gui

import extracells.container.ContainerHardMEDrive
import extracells.registries.BlockEnum
import extracells.tileentity.TileEntityHardMeDrive
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.inventory.GuiContainer
import net.minecraft.client.renderer.GlStateManager
import net.minecraft.entity.player.InventoryPlayer
import net.minecraft.inventory.Slot
import net.minecraft.util.ResourceLocation

class GuiHardMEDrive(inventory: InventoryPlayer, tile: TileEntityHardMeDrive)
  extends GuiContainer(new ContainerHardMEDrive(inventory, tile)) {


  xSize = 176
  ySize = 166
  private val guiTexture = new ResourceLocation("extracells", "textures/gui/hardmedrive.png")

  override def drawGuiContainerBackgroundLayer(f: Float, i: Int, j: Int) = {
    drawDefaultBackground();
    GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
    Minecraft.getMinecraft().renderEngine.bindTexture(guiTexture);
    val posX = (width - xSize) / 2;
    val posY = (height - ySize) / 2;
    drawTexturedModalRect(posX, posY, 0, 0, xSize, ySize);
    import scala.collection.JavaConversions._
    for (s <- this.inventorySlots.inventorySlots) {
      renderBackground(s.asInstanceOf[Slot])
    }
  }

  override def drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
    super.drawScreen(mouseX, mouseY, partialTicks)
    renderHoveredToolTip(mouseX, mouseY)
  }

  override def drawGuiContainerForegroundLayer(i: Int, j: Int) =
    fontRenderer.drawString(BlockEnum.BLASTRESISTANTMEDRIVE.getStatName, 5, 5, 0x000000)

  private def renderBackground(slot: Slot) {
    if ((slot.getStack == null || slot.getStack.isEmpty) && slot.slotNumber < 3) {
      GlStateManager.disableLighting()
      GlStateManager.enableBlend()
      GlStateManager.color(1.0F, 1.0F, 1.0F, 0.5F)
      this.mc.getTextureManager.bindTexture(new ResourceLocation("appliedenergistics2", "textures/guis/states.png"))
      this.drawTexturedModalRect(this.guiLeft + slot.xPos, this.guiTop + slot.yPos, 240, 0, 16, 16)
      GlStateManager.disableBlend()
      GlStateManager.enableLighting()
    }
  }

}
